### PR TITLE
Fixes to CodeQL warnings

### PR DESCRIPTION
**Pull request by nh758**
## Release Notes

- fix CodeQL Warnings:
   - https://github.com/digi-serve/ab_platform_web/security/code-scanning/576
   - https://github.com/digi-serve/ab_platform_web/security/code-scanning/552
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/490
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/491
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/59
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/61
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/422
  - https://github.com/digi-serve/ab_platform_web/security/code-scanning/566

 

## Test Status

behavior should be covered in existing tests

### CODE REVIEW

File: AppBuilder/ABFactory.js [hunk 1]
```diff
@@ -854,7 +854,7 @@ class ABFactory extends ABFactoryCore {
          console.error(message);
       }
 
-      if (rest && rest.length > 0) {
+      if (rest.length > 0) {
```
Comment (Hiro-Nakamura) [inline AppBuilder/ABFactory.js:857]: isn't the point of adding a `rest && rest.length` check to ensure `rest` is a valid value that we can then eval `.length` from?  

Like is it possible `rest` might be `undefined`, `null`,  or some other value that is `falsey`?  If so, doing a `rest.length` is a crash waiting to happen.

File: AppBuilder/platform/views/viewComponent/ABViewFormDatepickerComponent.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ module.exports = class ABViewFormDatepickerComponent extends (
          _ui.value = new Date(_ui.value);
 
       // if we have webix locale set, will use the date format form there.
-      if (field !== null && !window.webixLocale) _ui.format = field.getFormat();
+      if (!window.webixLocale) _ui.format = field.getFormat();
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/views/viewComponent/ABViewFormDatepickerComponent.js:74]: Again, same issue with `field` being `undefined` or `null`.  without that check, `field.getFormat()` will crash.

File: AppBuilder/platform/views/viewComponent/ABViewGridComponent.js [hunk 3]
```diff
@@ -254,9 +254,11 @@ export default class ABViewGridComponent extends ABViewComponent {
...
                }
             },
-            onBeforeEditStart: function (id) {
-               if (!this.getItem(id) == "appbuilder_select_item") return false;
-            },
+            // onBeforeEditStart: function (/*id*/) {
+            //    // Not sure what this is suposed to check, but this condition
+            //    // will always be false.
+            //    if (!this.getItem(id) == "appbuilder_select_item") return false;
+            // },
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/views/viewComponent/ABViewGridComponent.js:261]: OK, this is odd.  I happened to see this in my GitHub client.  It shows this modification as: 
```
-            onBeforeEditStart: function (id) {
-               if (this.getItem(id) !== "appbuilder_select_item") return false;
-            },
+            // onBeforeEditStart: function (/*id*/) {
+            //    // Not sure what this is suposed to check, but this condition
+            //    // will always be false.
+            //    if (!this.getItem(id) == "appbuilder_select_item") return false;
+            // },
```

The ` if (this.getItem(id) !== "appbuilder_select_item")` seems like a valid check to make sure that the value in `this.getItem(id)` isn't `appbuilder_select_item`.

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
Just a couple of questions about how we handle checking the validity of some of our variables.

File: AppBuilder/platform/views/viewComponent/ABViewFormDatepickerComponent.js (hunk 2 above)
Comment (nh758) [reply AppBuilder/platform/views/viewComponent/ABViewFormDatepickerComponent.js:74]: We already return `if (!field)` on line 57

File: AppBuilder/platform/views/viewComponent/ABViewGridComponent.js (hunk 3 above)
Comment (nh758) [reply AppBuilder/platform/views/viewComponent/ABViewGridComponent.js:261]: I did try that, but I believe it broke editing some things in the Grid. My suspicion is that this is left over from when we were using `selectivity`?

**Review by Hiro-Nakamura — APPROVED** [review]
LGTM

**Pull request merged**